tchannel-core/src/main/java/com/uber/tchannel/tracing/Tracing.java

```diff
@@ -89,7 +91,18 @@ private Tracing() {}
             }
         }
 
-        // TODO if tracer is Zipkin compatible, inject Trace fields
+        // if Jaeger span context, set Trace fields
+        if (span.context() instanceof JaegerSpanContext) {
+            JaegerSpanContext jaegerSpanContext = (JaegerSpanContext) span.context();
+            Trace trace = new Trace(
+                jaegerSpanContext.getSpanId(),
+                jaegerSpanContext.getParentId(),
+                // tchannel only support 64bit IDs, https://github.com/uber/tchannel/blob/master/docs/protocol.md#tracing
+                jaegerSpanContext.getTraceIdLow(),
+                jaegerSpanContext.getFlags());
+            request.setTrace(trace);
+        }
+
         // if request has headers, inject tracing context
         if (request instanceof TraceableRequest) {
             TraceableRequest traceableRequest = (TraceableRequest) request;
```

Comment:
Should we also try to extract trace from the headers and inject into trace field?

Reply:
This is actually the outbound path, where the headers don't have the trace info yet. The existing code in this method creates a child span and injects it into headers, what I added here is to also inject the same trace info into request trace fields.